test/helper.js

```diff
@@ -68,9 +68,9 @@ function startBroker (args) {
   })
 }
 
-function receiveMessage (receiver) {
+function receiveMessage (receiver, wait) {
   return new Promise((resolve, reject) => {
-    var timeout = setTimeout(reject.bind(Error('Timeout')), 500)
+    var timeout = setTimeout(reject.bind(this, Error('Timeout')), wait || 500)
```

Comment:
The of setTimeout cause issues. I'm sorry I did not spot this earlier, it should be replaced/removed.

Reply:
I sincerly don't know how to make it working without it

Reply:
500 is possibly too low depending on the environment :/. All this suite relies heavily on timeouts, and this makes it brittle/flaky.

Reply:
Ok, so I could remove that but in the case for example of subscriptions that I should not receive, how could I say that is not received and pass the test? Actually the timeout is mostly for that, for others I could simply remove that and wait that test timeout make the test fail in case message is not received